src/ray/gcs/gcs_server/gcs_resource_manager.cc

```diff
@@ -38,6 +38,12 @@ void GcsResourceManager::ConsumeSyncMessage(
   // delegate the work to the main thread for thread safety.
   // Ideally, all public api in GcsResourceManager need to be put into this
   // io context for thread safety.
+
```

Comment:
Or we could remove this: 

https://github.com/ray-project/ray/blob/3f11cf4666b4042f9ef66c3ee51486e5ece3cb94/src/ray/gcs/gcs_server/gcs_server.cc#L537-L538

Open to either.

Reply:
Good point. Let Yi make a call

Reply:
I'd prefer removing the register ones and put a RAY_CHECK here.

Reply:
Sounds good to me!